### ISSUE TITLE
Add custom attributes from the client

### ISSUE THREAD

**Issue by mlcollard** [srcML/srcML#1924]
We have the ability to declare a custom namespace in the client:

```console
srcml --text="a;" -l C++ --xmlns:metric="https://mlcollard.net/metric"
```

```XML
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<unit xmlns="http://www.srcML.org/srcML/src" xmlns:metric="https://mlcollard.net/metric" revision="1.0.0" language="C++"><expr_stmt><expr><name>a</name></expr>;</expr_stmt></unit>
```

But we have no way through the client to add the value of a custom, namespace attribute to an archive nor a unit, so it is not possible to, for example, add an attribute `metric:complexity` nor add a value, as in the following (scroll right to view added attribute):

```XML
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<unit xmlns="http://www.srcML.org/srcML/src" xmlns:metric="https://mlcollard.net/metric" revision="1.0.0" language="C++" metric:complexity="1"><expr_stmt><expr><name>a</name></expr>;</expr_stmt></unit>
```

To add this, we need an option that contains:

* PREfix or URI of namespace
* NAME of attribute
* VALUE of attribute

We actually have an existing way to add an attribute for an XPath query result:

```console
  --attribute PRE:NAME="VALUE"  Insert attribute PRE:NAME="VALUE" into element results of XPath query in original unit
```

so this form could be expanded and use for an additional purpose.  However, there is another form with matches how we declare additional namespaces as we saw above for the `--xmlns` option:

```console
srcml --text="a;" -l C++ --xmlns:metric="https://mlcollard.net/metric"
```

This borrows the same form as it exists in XML. We could do the following:

```console
srcml --text="a;" -l C++ --xmlns:metric="https://mlcollard.net/metric" --metric:complexity="1"
```

Note that to use the prefix `metric` the namespace must be declared in the command. So the prefix of each custom namespace forms a set of options as the xmlns does for namespaces, and the result should be:

```XML
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<unit xmlns="http://www.srcML.org/srcML/src" xmlns:metric="https://mlcollard.net/metric" revision="1.0.0" language="C++" metric:complexity="1"><expr_stmt><expr><name>a</name></expr>;</expr_stmt></unit>
```

Multiple custom attributes could be added:

```console
srcml --text="a;" -l C++ --xmlns:metric="https://mlcollard.net/metric" --metric:complexity="1" --metric:dit="0"
```

scroll right to view added attributes:

```XML
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<unit xmlns="http://www.srcML.org/srcML/src" xmlns:metric="https://mlcollard.net/metric" revision="1.0.0" language="C++" metric:complexity="1" metric:dit="0"><expr_stmt><expr><name>a</name></expr>;</expr_stmt></unit>
```

To summarize:
* Already have the ability to add custom namespaces
* Have an option `--attribute` for setting custom attributes on XPath results
* Extend the option `--attribute` to set general attributes
* Extend the form for custom options for each custom namespace, e.g., `--metric:dit="0"`

This issue depends on #1923. However there is another issue. The above examples are all solitary, non-archive srcML. In the case of an archive, the problem is more complex and similar to the standard attribute settings. For example, the following is allowed:

```console
srcml --text="a;" -l C++ --filename="a.cpp"
```

However, when used on an archive, it is not:

```console
$ srcml --text="a;" -l C++ --filename="a.cpp" --archive
srcml: --filename cannot be set for a srcML archive. Use --url instead
```

as we can only allow setting attributes on the root unit, not an individual unit. The issue is that in an archive we can have multiple units, and what does this option then mean? The first unit only? All the units? And there is no way to indicate the filename for multiple units via the command-line options.

One solution for this is the header format for null-separated source code in #1906 where each input (in this case text) could specify a specific attribute value for each unit, and even for custom attributes:

```YAML
---
    documentType: srcML
    filename: "a.cpp"
    "xmlns:metric": "https://mlcollard.net/metric"
    "metric:complexity": "1"
---
int n;
---
    documentType: srcML
    filename: "b.cpp"
    "xmlns:metric": "https://mlcollard.net/metric"
    "metric:complexity": "2"
---
if (n < 0)
    int m;
```   

The special character is the null-separator. The output would have the correct attribute value for each unit (scroll right to view added attribute):

```XML
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<unit xmlns="http://www.srcML.org/srcML/src" revision="1.0.0">

<unit xmlns:metric="https://mlcollard.net/metric" revision="1.0.0" language="C" filename="a.cpp" hash="d00123a072007a932bc99866fe4c435715e514ef" metric:complexity="1"><decl_stmt><decl><type><name>int</name></type> <name>n</name></decl>;</decl_stmt>
</unit>

<unit xmlns:metric="https://mlcollard.net/metric" revision="1.0.0" language="C" filename="b.cpp" hash="5386e51a84448ea2f3b9be6b40b44907738dc3b0" metric:complexity="2"><if_stmt><if>if <condition>(<expr><name>n</name> <operator>&lt;</operator> <literal type="number">0</literal></expr>)</condition><block type="pseudo"><block_content>
    <decl_stmt><decl><type><name>int</name></type> <name>m</name></decl>;</decl_stmt></block_content></block></if></if_stmt>
</unit>

</unit>
```
